### ISSUE TITLE
0.9.41 - > hs.mouse.getCurrentScreen() fails

### ISSUE THREAD

**Issue by bradparks** [Hammerspoon/hammerspoon#570]
Hey there! I just upgraded to Hammerspoon 0.9.41, and immediately noticed that the following method seems to be failing now

`hs.mouse.getCurrentScreen()`

It gives the following error when run via an extension, or run in the console:

```
> hs.mouse.getCurrentScreen()
-- Loading extension: mouse
...n.app/Contents/Resources/extensions/hs/geometry/init.lua:573: attempt to compare nil with number
```

I tried it on a system with both a single monitor (Macbook Air) and one with 2 additional external monitors and got the same result. Is there a better way to report a dump or bug?

**Comment by cmsj** [comment]
damn!

**Comment by cmsj** [comment]
(I mean, thanks, good catch. I'm just annoyed that we released with a silly bug :)

**Comment by bradparks** [comment]
hey... no worries! I'm loving Hammerspoon so I'm very understanding.... thanks again! and I'm sure you've found it, but it looks to be at this line to me:

https://github.com/Hammerspoon/hammerspoon/blob/ff5d964dff657f75b5aeba524816b65bd455f315/extensions/geometry/init.lua#L573

with the `x2` and `y2` properties, i think....

**Comment by cmsj** [comment]
@lowne looks like the parameters might be reversed here? Although I can't get it to return a value even if I do that.

**Comment by lowne** [comment]
Whops. My bad, sorry everyone :/
#571 for the fix if urgent. I'll try to audit the rest of geometry to see if that commit (12ab412) had other unintended consequences.

**Comment by lowne** [comment]
Unfortunately it appears there's no ready function for a temporary workaround 😭 - I'll add the possibility to pass arbitrary points and rects to screen.find (so that e.g. `hs.mouse.getCurrentScreen()` could be also done with `hs.screen.find(hs.mouse.getAbsolutePosition())`)

**Comment by cmsj** [comment]
@lowne hmm, even with #571 applied, getCurrentScreen() still seems to fail on a nil/number comparison

**Comment by bradparks** [comment]
@lowne - I'm finding the same as @cmsj, that it works in some situations, and not in others.... so this is still broken in some way at some times. I'm getting around it for the time being by using `hs.screen.primaryScreen()`....

**Comment by lowne** [comment]
works fine here, can I have more details (and stack traces)?

**Comment by bradparks** [comment]
hmmm.. i don't seem to be able to duplicate it.... i'll definitely let you know if I do... thanks for looking into it...

**Comment by oskarols** [comment]
0.9.41

> hs.mouse.getCurrentScreen()
...n.app/Contents/Resources/extensions/hs/geometry/init.lua:573: attempt to compare nil with number

**Comment by lowne** [comment]
For a temporary patch until next release, put this (which basically "applies" #571) at the *top* of your `init.lua`:
```lua
-- temporary patch for #570
hs.geometry.isPointInRect=function(t1,...)
   t1=hs.geometry.new(t1) local t2=hs.geometry.new(...)
  return t1.x>=t2.x and t1.y>=t2.y and (t1.x2 or t1.x)<=(t2.x2 or t2.x) and (t1.y2 or t1.y)<=(t2.y2 or t2.y)
 end
```
@oskarols let me know if it works

**Comment by cmsj** [comment]
I was wrong, this does seem to be fixed by #571. Thanks @lowne :)

**Comment by bradparks** [comment]
yeah @lowne - thanks! :trophy:
